qiskit/circuit/controlflow/if_else.py

```diff
@@ -448,7 +448,8 @@ def __enter__(self):
             raise CircuitError("Cannot attach an 'else' to a broadcasted 'if' block.")
         appended = appended_instructions[0]
         instruction = circuit._peek_previous_instruction_in_scope()
-        if appended is not instruction:
+        # TODO: this isn't exactly the right check
+        if appended.operation is not instruction.operation:
```

Comment:
It's not precisely the same, but still good enough for the context.  If a user is screwing with the circuit so heavily that this check fails to trigger the error but somehow the circuit isn't valid, there's plenty that they could have done to screw everything up anyway (mutating the operation in place, for example).

Reply:
Sounds good. I was mostly just concerned for the future when the singleton stuff lands. At that point, this check will be less useful.

Reply:
This one will still be needed. The singleton gates won't allow control-flow operations to become singletons (this `appended.operation` is always an `IfElseOp`, iirc).